File: src/3427.sum-of-variable-length-subarrays.3731/solution.ts
function subarraySum (nums: number[]): number {
    let result = 0;
    const prefixs = new Array<number>(nums.length + 1);
    prefixs[0] = 0;
    for (let i = 0; i < nums.length; i++) {
        prefixs[i + 1] = prefixs[i] + nums[i];
        const start = Math.max(0, i - nums[i]);
        result += prefixs[i + 1] - prefixs[start];
    }
    return result;
}

File: src/3427.sum-of-variable-length-subarrays.3731/solution.java
class Solution {
    public int subarraySum(int[] nums) {
        int result = 0;
        int[] prefixs = new int[nums.length+1];
        for(int i=0;i< nums.length;i++){
            prefixs[i+1] = prefixs[i]+nums[i];
            int start = Math.max(0,i-nums[i]);
            result += prefixs[i+1]-prefixs[start];
        }
        return result;
    }
}